### ISSUE TITLE
Blank sources generated during test.

### ISSUE THREAD

**Issue by leesdolphin** [timvideos/gst-switch#182]
The following 2 images were produced during [this build](https://travis-ci.org/leesdolphin/gst-switch/jobs/47902990#L3420)
![Example 1](http://i.imgur.com/rGarR8I.png) ![Example 2](http://i.imgur.com/rLcuutl.png)

The part of the logs that matter:
```
tests/integrationtests/test_controller.py::TestSetCompositeMode::test_set_composite_mode_equal Starting server
Creating process ['../tools/gst-switch-srv', '--video-input-port=3000', '--audio-input-port=4000', '--control-port=5000', '--video-format=debug']
['../tools/gst-switch-srv', '--video-input-port=3000', '--audio-input-port=4000', '--control-port=5000', '--video-format=debug']
start preview

video/x-raw,
  format=(string)I420, pixel-aspect-ratio=(fraction)1/1,
  width=(int)300, height=(int)200,
  framerate=(fraction)25/1


video/x-raw,
  format=(string)I420, pixel-aspect-ratio=(fraction)1/1,
  width=(int)300, height=(int)200,
  framerate=(fraction)25/1

False
end preview
TESTS: [<gstswitch.testsource.VideoSrc object at 0x2b623b3cd890>, <gstswitch.testsource.VideoSrc object at 0x2b623b3cd8d0>]
End source with pattern 4
End source with pattern 5
Killing server
GCOV FLUSH
/home/travis/build/leesdolphin/gst-switch/tools/
I was going to do some coverage stuff. But thought better of it
Server Killed
/tmp/tmpNhS7Gz
avconv version 9.16-6:9.16-0ubuntu0.14.04.1, Copyright (c) 2000-2014 the Libav developers
  built on Aug 10 2014 18:16:02 with gcc 4.8 (Ubuntu 4.8.2-19ubuntu1)
  configuration: --arch=amd64 --enable-pthreads --enable-runtime-cpudetect --extra-version='6:9.16-0ubuntu0.14.04.1' --libdir=/usr/lib/x86_64-linux-gnu --prefix=/usr --enable-bzlib --enable-libdc1394 --enable-libfreetype --enable-frei0r --enable-gnutls --enable-libgsm --enable-libmp3lame --enable-librtmp --enable-libopenjpeg --enable-libopus --enable-libpulse --enable-libschroedinger --enable-libspeex --enable-libtheora --enable-vaapi --enable-vdpau --enable-libvorbis --enable-libvpx --enable-zlib --enable-gpl --enable-swscale --enable-libcdio --enable-x11grab --enable-libx264 --enable-libxvid --shlibdir=/usr/lib/x86_64-linux-gnu --enable-shared --disable-static
  libavutil     52.  3. 0 / 52.  3. 0
  libavcodec    54. 35. 0 / 54. 35. 0
  libavformat   54. 20. 4 / 54. 20. 4
  libavdevice   53.  2. 0 / 53.  2. 0
  libavfilter    3.  3. 0 /  3.  3. 0
  libavresample  1.  0. 1 /  1.  0. 1
  libswscale     2.  1. 1 /  2.  1. 1
[mjpeg @ 0x165eb40] mjpeg: JFIF header found (version: 1.1) SAR=1/1
Input #0, avi, from 'output-3.data':
  Duration: 00:00:00.00, start: 0.000000, bitrate: -2147483 kb/s
    Stream #0.0: Video: mjpeg, yuvj420p, 300x200 [PAR 1:1 DAR 3:2], 30 fps, 30 tbr, 30 tbn
[buffer @ 0x16b5160] w:300 h:200 pixfmt:yuvj420p
[buffersink @ 0x165e320] auto-inserting filter 'auto-inserted fifo 0' between the filter 'format' and the filter 'output stream 0:0'
[format @ 0x165e640] auto-inserting filter 'auto-inserted scaler 0' between the filter 'Parsed filter 0 null' and the filter 'format'
[scale @ 0x166d360] w:300 h:200 fmt:yuvj420p -> w:300 h:200 fmt:rgb24 flags:0x4
Output #0, image2, to '/tmp/tmpNhS7Gz/out3_1.png':
  Metadata:
    encoder         : Lavf54.20.4
    Stream #0.0: Video: png, rgb24, 300x200 [PAR 1:1 DAR 3:2], q=2-31, 200 kb/s, 90k tbn, 30 tbc
Stream mapping:
  Stream #0:0 -> #0:0 (mjpeg -> png)
Press ctrl-c to stop encoding
[mjpeg @ 0x165eb40] mjpeg: JFIF header found (version: 1.1) SAR=1/1
    Last message repeated 60 times
No more output streams to write to, finishing.
Bitrate not available, avio_tell() failed: Invalid argument
frame=    1 fps=  0 q=0.0 Lsize=       0kB time=0.03 bitrate=   0.0kbits/s    

video:2kB audio:0kB global headers:0kB muxing overhead -100.000000%
avconv version 9.16-6:9.16-0ubuntu0.14.04.1, Copyright (c) 2000-2014 the Libav developers
  built on Aug 10 2014 18:16:02 with gcc 4.8 (Ubuntu 4.8.2-19ubuntu1)
  configuration: --arch=amd64 --enable-pthreads --enable-runtime-cpudetect --extra-version='6:9.16-0ubuntu0.14.04.1' --libdir=/usr/lib/x86_64-linux-gnu --prefix=/usr --enable-bzlib --enable-libdc1394 --enable-libfreetype --enable-frei0r --enable-gnutls --enable-libgsm --enable-libmp3lame --enable-librtmp --enable-libopenjpeg --enable-libopus --enable-libpulse --enable-libschroedinger --enable-libspeex --enable-libtheora --enable-vaapi --enable-vdpau --enable-libvorbis --enable-libvpx --enable-zlib --enable-gpl --enable-swscale --enable-libcdio --enable-x11grab --enable-libx264 --enable-libxvid --shlibdir=/usr/lib/x86_64-linux-gnu --enable-shared --disable-static
  libavutil     52.  3. 0 / 52.  3. 0
  libavcodec    54. 35. 0 / 54. 35. 0
  libavformat   54. 20. 4 / 54. 20. 4
  libavdevice   53.  2. 0 / 53.  2. 0
  libavfilter    3.  3. 0 /  3.  3. 0
  libavresample  1.  0. 1 /  1.  0. 1
  libswscale     2.  1. 1 /  2.  1. 1
[mjpeg @ 0x239db40] mjpeg: JFIF header found (version: 1.1) SAR=1/1
Input #0, avi, from 'output-3.data':
  Duration: 00:00:00.00, start: 0.000000, bitrate: -2147483 kb/s
    Stream #0.0: Video: mjpeg, yuvj420p, 300x200 [PAR 1:1 DAR 3:2], 30 fps, 30 tbr, 30 tbn
[buffer @ 0x23f4160] w:300 h:200 pixfmt:yuvj420p
[buffersink @ 0x239d320] auto-inserting filter 'auto-inserted fifo 0' between the filter 'format' and the filter 'output stream 0:0'
[format @ 0x239d640] auto-inserting filter 'auto-inserted scaler 0' between the filter 'Parsed filter 0 null' and the filter 'format'
[scale @ 0x23ac360] w:300 h:200 fmt:yuvj420p -> w:300 h:200 fmt:rgb24 flags:0x4
Output #0, image2, to '/tmp/tmpNhS7Gz/out3_2.png':
  Metadata:
    encoder         : Lavf54.20.4
    Stream #0.0: Video: png, rgb24, 300x200 [PAR 1:1 DAR 3:2], q=2-31, 200 kb/s, 90k tbn, 30 tbc
Stream mapping:
  Stream #0:0 -> #0:0 (mjpeg -> png)
Press ctrl-c to stop encoding
[mjpeg @ 0x239db40] mjpeg: JFIF header found (version: 1.1) SAR=1/1
    Last message repeated 150 times
No more output streams to write to, finishing.
Bitrate not available, avio_tell() failed: Invalid argument
frame=    1 fps=  0 q=0.0 Lsize=       0kB time=0.03 bitrate=   0.0kbits/s    

video:2kB audio:0kB global headers:0kB muxing overhead -100.000000%
RESULTS 0.542455555556 0.542455555556
./imgurbash.sh /tmp/tmpNhS7Gz/*.*
http://i.imgur.com/rGarR8I.png
Delete page: http://imgur.com/delete/RuvBMwJUJnJI8tt
http://i.imgur.com/rLcuutl.png
Delete page: http://imgur.com/delete/QyLThjQtT0aWkVn
0
```
...
```
______________ TestSetCompositeMode.test_set_composite_mode_equal ______________

self = <integrationtests.test_controller.TestSetCompositeMode object at 0x2b623b3cd6d0>

    def test_set_composite_mode_equal(self):
        """Test set_composite_mode"""
>       self.set_composite_mode(Controller.COMPOSITE_DUAL_EQUAL)

tests/integrationtests/test_controller.py:390: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <integrationtests.test_controller.TestSetCompositeMode object at 0x2b623b3cd6d0>
mode = 3, generate_frames = False

    def set_composite_mode(self, mode, generate_frames=False):
        """Create Controller object and call set_composite_mode method"""
        for _ in range(self.NUM):
    
            serv = Server(path=PATH, video_format="debug")
            try:
                serv.run()
    
                preview = PreviewSinks()
                preview.run()
    
                out_file = 'output-{0}.data'.format(mode)
                video_sink = VideoFileSink(serv.video_port + 1, out_file)
    
                sources = TestSources(video_port=3000)
                sources.new_test_video(pattern=4)
                sources.new_test_video(pattern=5)
    
                time.sleep(3)
                # expected_result = [mode != 3] * self.FACTOR
                # print(mode, expected_result)
                controller = Controller()
                res = controller.set_composite_mode(mode)
                print(res)
                time.sleep(3)
                video_sink.terminate()
                preview.terminate()
                sources.terminate_video()
                serv.terminate(1)
                if not generate_frames:
                    controller = Controller()
                    if mode == Controller.COMPOSITE_DUAL_EQUAL:
                        assert res is False
                    else:
                        assert res is True
>                   assert self.verify_output(mode, out_file) is True
E                   assert <bound method TestSetCompositeMode.verify_output of <integrationtests.test_controller.TestSetCompositeMode object at 0x2b623b3cd6d0>>(3, 'output-3.data') is True
E                    +  where <bound method TestSetCompositeMode.verify_output of <integrationtests.test_controller.TestSetCompositeMode object at 0x2b623b3cd6d0>> = <integrationtests.test_controller.TestSetCompositeMode object at 0x2b623b3cd6d0>.verify_output
```

**Comment by MaZderMind** [comment]
I'm pretty sure that this is equal to #141.
In short: The default resolution of the server changed to 720p, the test-sources still generate 300x200.
hyades started working on this: https://github.com/hyades/gst-switch/tree/720p-frames
But I also saw some work being done on aut-generating the test-frames soring the test preparation
(I personally don't like the test depending on things they generate on the test-target, though)

**Comment by MaZderMind** [comment]
fixed in #168
